src/vm/database/clarity_db.rs

```diff
@@ -102,27 +101,45 @@ impl <'a> ClarityDatabase <'a> {
         format!("vm::{}::{}::{}", contract_identifier, data as u8, var_name)
     }
 
+    pub fn make_metadata_key(data: StoreType, var_name: &str) -> String {
+        format!("vm-metadata::{}::{}", data as u8, var_name)
+    }
+
     pub fn make_key_for_quad(contract_identifier: &QualifiedContractIdentifier, data: StoreType, var_name: &str, key_value: String) -> String {
         format!("vm::{}::{}::{}::{}", contract_identifier, data as u8, var_name, key_value)
     }
 
-    pub fn make_contract_key(contract_identifier: &QualifiedContractIdentifier) -> String {
-        format!("vm::{}::{}", contract_identifier, StoreType::Contract as u8)
+    pub fn insert_contract_hash(&mut self, contract_identifier: &QualifiedContractIdentifier, contract_content: &str) -> Result<()> {
+        let hash = Sha512Trunc256Sum::from_data(contract_content.as_bytes()).0;
```

Comment:
In this scheme, two different contracts can be instantiated with the same code body, and have the same contract hash.  Is this what we want, or do we want the contract hash to be unique to the (contract-identifier, contract-code) pair?

Reply:
I don't think it matters either way --- the metadata is keyed by (blockhash, contract-identifier), so the contract hash isn't used for that. And the entry for the contract hash in the MARF is keyed by contract-identifier, so they're distinct commitments there too.